auditlog/models.py

```diff
@@ -66,26 +66,6 @@ def log_create(self, instance, force_log: bool = False, **kwargs):
             if callable(get_additional_data):
                 kwargs.setdefault("additional_data", get_additional_data())
 
-            # Delete log entries with the same pk as a newly created model.
-            # This should only be necessary when a pk is used twice.
-            if kwargs.get("action", None) is LogEntry.Action.CREATE:
-                if (
-                    kwargs.get("object_id", None) is not None
-                    and self.filter(
-                        content_type=kwargs.get("content_type"),
-                        object_id=kwargs.get("object_id"),
-                    ).exists()
-                ):
-                    self.filter(
-                        content_type=kwargs.get("content_type"),
-                        object_id=kwargs.get("object_id"),
-                    ).delete()
-                else:
-                    self.filter(
-                        content_type=kwargs.get("content_type"),
-                        object_pk=kwargs.get("object_pk", ""),
-                    ).delete()
-
```

Comment:
Why was this added in the first place?
https://thoughtbot.com/blog/chestertons-fence

Reply:
The primary key reuse is still a thing in mysql ([link](https://dev.mysql.com/doc/refman/8.0/en/example-auto-increment.html#:~:text=In%20this%20case%20(when%20the,values%20normally%20are%20not%20reused.))). Should we handle this by checking the db engine? Or maybe by a configuration var?

Reply:
@aqeelat, I don't think we should *ever* delete auditlog entries by default. The audit log exists to be immutable.

Reply:
@aleh-rymasheuski I agree. However, people who are serious about auditlogs will also be careful to avoid primary-key reuse.
The main issue is that when accessing the logs for a instance, we shouldn't get entries for older instances with the same object id.
I know this is an edge case, which is why I'm trying to find the most inclusive solution.

Reply:
Thanks @aqeelat for your input here. 

IMHO, we can merge it.